### PR TITLE
Improve app/daemon development

### PR DESCRIPTION
**Pull request by seanyesmunt**
- No address validation with `NO_ADDRESS_VALIDATION` env
- No daemon mismatch modal when running in dev mode
- Keep the app open if something crashes when running in dev mode

Waiting on testing from @eukreign

### CODE REVIEW

File: src/main/index.js [hunk 1]
```diff
@@ -59,16 +59,18 @@ app.on('ready', async () => {
...
     daemon.on('exit', () => {
-      daemon = null;
-      if (!appState.isQuitting) {
-        dialog.showErrorBox(
-          'Daemon has Exited',
-          'The daemon may have encountered an unexpected error, or another daemon instance is already running. \n\n' +
-            'For more information please visit: \n' +
-            'https://lbry.io/faq/startup-troubleshooting'
-        );
+      if (isDev) {
```
Comment (eukreign) [inline src/main/index.js]: Should this be `!isDev`?

File: src/renderer/redux/actions/app.js [hunk 2]
```diff
@@ -253,17 +253,19 @@ export function doCheckUpgradeSubscribe() {
 export function doCheckDaemonVersion() {
   return dispatch => {
     Lbry.version().then(({ lbrynet_version: lbrynetVersion }) => {
-      if (config.lbrynetDaemonVersion === lbrynetVersion) {
-        dispatch({
+      // Avoid the incompatible daemon modal if running in dev mode
+      // Lets you run a different daemone than the one specificed in package.json
```
Comment (eukreign) [inline src/renderer/redux/actions/app.js]: typo `daemone` and `specificed`

Comment (seanyesmunt) [reply src/renderer/redux/actions/app.js]: nice

File: src/main/index.js (hunk 1 above)
Comment (seanyesmunt) [reply src/main/index.js]: whoops

File: src/main/index.js [hunk 3]
```diff
@@ -59,16 +59,18 @@ app.on('ready', async () => {
...
-          'The daemon may have encountered an unexpected error, or another daemon instance is already running. \n\n' +
-            'For more information please visit: \n' +
-            'https://lbry.io/faq/startup-troubleshooting'
-        );
+      if (!isDev) {
+        daemon = null;
+        if (!appState.isQuitting) {
+          dialog.showErrorBox(
+            'Daemon has Exited',
+            'The daemon may have encountered an unexpected error, or another daemon instance is already running. \n\n' +
```
Comment (skhameneh) [inline src/main/index.js]: Template literals might help with formatting here?

**Pull request merged**